### PR TITLE
Libdrm

### PR DESCRIPTION
**Pull request by ericLemanissier**
alternative to https://github.com/conan-io/conan-center-index/pull/8302

fixes conan-io/conan-center-index#8024

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libdrm/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.intel:
+            self.requires("xorg/system")
```
Comment (madebr) [inline recipes/libdrm/all/conanfile.py:77]: When disabling intel, will it work on Windows?
I would be surprised by that.

Comment (ericLemanissier) [reply recipes/libdrm/all/conanfile.py:77]: It does not. I need to raise an exception

Comment (madebr) [reply recipes/libdrm/all/conanfile.py:77]: Another question, does this recipe need `kernel-headers-generic`?

Comment (ericLemanissier) [reply recipes/libdrm/all/conanfile.py:77]: I have no idea!

Comment (ericLemanissier) [reply recipes/libdrm/all/conanfile.py:77]: How can I find this information?

Comment (madebr) [reply recipes/libdrm/all/conanfile.py:77]: Analyzing the rpm scripts of distributions, or browsing the sources.
I found this in the headers:
https://gitlab.freedesktop.org/mesa/drm/-/blob/main/include/drm/drm.h#L38-L41
I think those are provided by the kernel headers.

**Review by madebr — DISMISSED** [review]

**Comment by ericLemanissier** [comment]
blocked by #8355

File: recipes/libdrm/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    def build(self):
+        meson = self._configure_meson()
+        meson.build()
+
+    def package(self):
+        meson = self._configure_meson()
+        meson.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.mkdir(os.path.join(self.package_folder, "licenses"))
+        tools.save(os.path.join(self.package_folder, "licenses", "TBD"), "TO BE DONE")
```
Comment (dvirtz) [inline recipes/libdrm/all/conanfile.py:118]:
```suggestion
        # Extract the License/s from the header to a file
        tmp = tools.load(os.path.join(self._source_subfolder, "include", "drm", "drm.h"))
        license_contents = re.search("\*\/.*(\/\*(\*(?!\/)|[^*])*\*\/)", tmp, re.DOTALL)[1]
        tools.save(os.path.join(self.package_folder, "licenses", "LICENSE"), license_contents)
```

File: recipes/libdrm/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+
+class LibdrmConan(ConanFile):
+    name = "libdrm"
+    description = "User space library for accessing the Direct Rendering Manager, on operating systems that support the ioctl interface"
+    topics = ("libdrm", "graphics")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://gitlab.freedesktop.org/mesa/drm"
+    license = "MIT"
+
```
Comment (dvirtz) [inline recipes/libdrm/all/conanfile.py:14]: generators = "PkgConfigDeps"

Comment (ericLemanissier) [inline recipes/libdrm/all/conanfile.py:14]:
```suggestion
    generators = "PkgConfigDeps"

```

File: recipes/libdrm/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        meson.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.mkdir(os.path.join(self.package_folder, "licenses"))
+        # Extract the License/s from the header to a file
+        tmp = tools.load(os.path.join(self._source_subfolder, "include", "drm", "drm.h"))
+        license_contents = re.search("\*\/.*(\/\*(\*(?!\/)|[^*])*\*\/)", tmp, re.DOTALL)[1]
+        tools.save(os.path.join(self.package_folder, "licenses", "LICENSE"), license_contents)
+
+    def package_info(self):
+        self.cpp_info.components["libdrm_libdrm"].libs = ["drm"]
```
Comment (dvirtz) [inline recipes/libdrm/all/conanfile.py:125]: what's the reason for the `libdrm_` prefix in component names?

Comment (ericLemanissier) [reply recipes/libdrm/all/conanfile.py:125]: this avoids collision with other recipes' components. Also, the component name cannot be identical to recipe name

Comment (dvirtz) [reply recipes/libdrm/all/conanfile.py:125]: aren't components scoped? I mean to access a component you need to write `self.deps_cpp_info[package_name].components[component_name]`. 
can you point me to an example of such colission?
re. the second point, you can name the `"libdrm_libdrm` component `headers`.

Comment (ericLemanissier) [reply recipes/libdrm/all/conanfile.py:125]: The components are scoped, but the files generated by generators (cmake_find_package etc.) are not

**Review by madebr — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**